src/Control/Optics/Linear/Internal.hs

```diff
@@ -53,6 +56,12 @@ _Left = Optical first
 _Right :: Prism a b (Either c a) (Either c b)
 _Right = Optical second
 
+_Just :: Prism a b (Maybe a) (Maybe b)
+_Just = prism Just (maybe (Left Nothing) Right)
```

Comment:
For the records: I discourage the temptation to use pointfree at all cost. This one is still ok though. But only just.

Conciseness is not always what you want to shoot for. Clarity is.